crates/sound/src/vhu_sound.rs

```diff
@@ -275,34 +286,38 @@ impl VhostUserSoundThread {
                         .memory()
                         .read_obj::<VirtioSndPcmSetParams>(desc_request.addr())
                         .map_err(|_| Error::DescriptorReadFailed)?;
-                    let stream_id: usize = u32::from(set_params.hdr.stream_id) as usize;
+
+                    let params = PCMParams {
+                        buffer_bytes :  set_params.buffer_bytes.into(),
+                        period_bytes : set_params.period_bytes.into(),
+                        features : set_params.features.into(),
+                        rate : set_params.rate.into(),
+                        format : set_params.format.into(),
+                        channels : set_params.channels.into()
+                    };
+
+                    let stream_id = u32::from(set_params.hdr.stream_id);
                     let buffer_bytes: u32 = set_params.buffer_bytes.into();
                     let period_bytes: u32 = set_params.period_bytes.into();
-                    let features: u32 = set_params.features.into();
 
-                    dbg!("stream_id: {}", stream_id );
-                    dbg!("set params format: {}", set_params.format);
-                    dbg!("set params rate: {} ", set_params.rate);
-                    dbg!("set params channels: {} ", set_params.channels);
-                    if features != 0 {
+                    if params.features != 0 {
                         error!("No feature is supported");
-                        desc_chain
-                            .memory()
-                            .write_obj(VIRTIO_SND_S_NOT_SUPP, desc_response.addr())
-                            .map_err(|_| Error::DescriptorWriteFailed)?;
-                    }
-                    if buffer_bytes % period_bytes != 0 {
+                        response = VirtioSoundHeader { code: VIRTIO_SND_S_NOT_SUPP.into(), };
+                    } else if buffer_bytes % period_bytes != 0 {
```

Comment:
i think this should be params.buffer_bytes and params.period_bytes no?

Reply:
Yes, thanks.